Common/src/Storage/SessionTableExtensions.cs

```diff
@@ -110,4 +112,169 @@ public static class SessionTableExtensions
                                          e);
     }
   }
+
+  /// <summary>
+  ///   Cancel a session
+  /// </summary>
+  /// <param name="sessionTable">Interface to manage sessions lifecycle</param>
+  /// <param name="sessionId">Id of the session to cancel</param>
+  /// <param name="cancellationToken">Token used to cancel the execution of the method</param>
+  /// <returns>
+  ///   The metadata of the cancelled session
+  /// </returns>
+  public static async Task<SessionData> CancelSessionAsync(this ISessionTable sessionTable,
```

Comment:
I think this should throw if the session is purged.

Reply:
That's already the case.